enf/resource_enf_network.go

```diff
@@ -0,0 +1,95 @@
+package enf
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"strconv"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/xaptum/go-enf/enf"
+)
+
+func resourceEnfNetwork() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceEnfNetworkCreate,
+		Read:   resourceEnfNetworkRead,
+		Update: resourceEnfNetworkUpdate,
+		Delete: resourceEnfFirewallRuleDelete,
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"description": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"status": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceEnfNetworkCreate(d *schema.ResourceData, meta interface{}) error {
+	domainID := meta.(*EnfClient).DomainID
+
+	params := &enf.NetworkRequest{
+		Name:        enf.String(d.Get("name").(string)),
+		Description: enf.String(d.Get("description").(string)),
+	}
+
+	client := meta.(*EnfClient).Client
+	network, _, err := client.Network.CreateNetwork(context.Background(), strconv.FormatInt(domainID, 10), params)
+	if err != nil {
+		return err
+	}
+
+	d.SetId(*network.Network)
+
+	return resourceEnfNetworkRead(d, meta)
+}
+
+func resourceEnfNetworkRead(d *schema.ResourceData, meta interface{}) error {
+	address := d.Id() // this is the network address
+
+	client := meta.(*EnfClient).Client
+	network, resp, err := client.Network.GetNetwork(context.Background(), address)
+	if err != nil {
+		if resp.StatusCode == http.StatusNotFound {
+			d.SetId("")
+			return nil
+		}
+		return err
+	}
+
+	d.Set("name", network.Name)
+	d.Set("description", network.Description)
+	d.Set("status", network.Status)
+
+	return nil
+}
+
+func resourceEnfNetworkUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*EnfClient).Client
+
+	if d.HasChange("name") || d.HasChange("description") {
+		fields := &enf.NetworkRequest{
+			Name:        enf.String(d.Get("name").(string)),
+			Description: enf.String(d.Get("description").(string)),
+		}
+		_, _, err := client.Network.UpdateNetwork(context.Background(), d.Id(), fields)
+		if err != nil {
+			return fmt.Errorf("Error updating network: %s", err)
+		}
+	}
+
+	return resourceEnfFirewallRuleRead(d, meta)
+}
+
+func resourceEnfNetworkDelete(d *schema.ResourceData, meta interface{}) error {
+	return nil
```

Comment:
Let's return an error here, since we don't support deleting.


As written, this lies to Terraform that the network was deleted, which will then lie to the user that the network was deleted.   I'd be upset to find out that all the resources I thought were deleted, actually weren't.

Just return an error, something like `Error("The network resource does not support deletion.")`.

Reply:
I don't see why this is crashing --- but let's get that figured out before merging the PR.

Reply:
Added error handling and stopped Terraform from crashing in latest commit.